zokrates_core/src/solvers/mod.rs

```diff
@@ -0,0 +1,147 @@
+use std::fmt;
+use zokrates_embed::generate_sha256_round_witness;
+use zokrates_field::field::Field;
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize, Hash, Eq)]
+pub enum Solver {
+    ConditionEq,
+    Bits,
+    Div,
+    Sha256Round,
+}
+
+impl fmt::Display for Solver {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl Signed for Solver {
+    fn get_signature(&self) -> (usize, usize) {
+        match self {
+            Solver::ConditionEq => (1, 2),
+            Solver::Bits => (1, 254),
```

Comment:
As this depends on the field, this should be the bitwidth of the prime p. Not this PR, but important.

Reply:
Yep, handled in #493